src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationOptions.cs

```diff
@@ -246,96 +212,30 @@ public OpenIdConnectProtocolValidator ProtocolValidator
         /// <summary>
         /// Gets or sets the type used to secure data handled by the middleware.
         /// </summary>
-        public ISecureDataFormat<AuthenticationProperties> StateDataFormat
-        {
-            get
-            {
-                return _stateDataFormat;
-            }
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException("value");
-                }
-
-                _stateDataFormat = value;
-            }
-        }
+        public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
 
         /// <summary>
         /// Gets or sets the type used to secure strings used by the middleware.
         /// </summary>
-        public ISecureDataFormat<string> StringDataFormat
-        {
-            get
-            {
-                return _stringDataFormat;
-            }
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException("value");
-                }
-
-                _stringDataFormat = value;
-            }
-        }
+        public ISecureDataFormat<string> StringDataFormat { get; set; }
 
         /// <summary>
         /// Gets or sets the <see cref="SecurityTokenValidators"/> for validating tokens.
         /// </summary>
         /// <exception cref="ArgumentNullException">if 'value' is null.</exception>
-        public ICollection<ISecurityTokenValidator> SecurityTokenValidators
-        {
-            get
-            {
-                return _securityTokenValidators;
-            }
-
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException("SecurityTokenValidators");
-                }
-
-                _securityTokenValidators = value;
-            }
-        }
+        public IList<ISecurityTokenValidator> SecurityTokenValidators { get; } = new List<ISecurityTokenValidator>(); 
```

Comment:
What I am trying to get across about order is that devs should not count on it, tends to be fragile. In fact it could be the case that two handlers are in the collection that can process the same token.
Not sure how to think about that matrix yet, but it is on my mind.